clinica/iotools/converters/adni_to_bids/adni_modalities/adni_fmap.py

```diff
@@ -245,165 +253,217 @@ def fmap_image(
+        lvl="info",
+    )
+
+    files = [f for f in os.listdir(fmap_path) if not f.startswith(".")]
+    js = [f for f in files if f.endswith("ph.json")][0]
+
+    # Checking for Echotime keys in phase json
+    with open(fmap_path / js, "r") as file:
+        json_data = json.load(file)
+    if "EchoTime1" not in json_data or "EchoTime2" not in json_data:
+        cprint(f'Invalid file {js}, missing "EchoTime" key.', lvl="warning")
+        unrecognized_fmap_case(fmap_path)
+        return
+    # Renaming
+    rename_files(fmap_path, files, BIDSFMAPCase.ONE_PHASE_TWO_MAGNITUDES)
+
+
+def two_phases_two_magnitudes_handler(fmap_path: Path):
+    "Performs the checks and renaming for BIDS spec case 2 for fieldmaps"
+
+    sub = fmap_path.parent.parent.name
+    ses = fmap_path.parent.name
+    cprint(
+        f"BIDS Case 2 : expecting 2 phase and 2 magnitude files for {sub}, {ses}.",
+        lvl="info",
+    )
+
+    files = [f for f in os.listdir(fmap_path) if not f.startswith(".")]
+    check_json = [f for f in files if "ph.json" in f]
+    # Checking for Echotime keys in phase jsons
+    for js in check_json:
+        with open(fmap_path / js, "r") as file:
+            json_data = json.load(file)
+        if "EchoTime" not in json_data:
+            cprint(f'Invalid file {js}, missing "EchoTime" key.', lvl="warning")
+            unrecognized_fmap_case(fmap_path)
+            return
+    # Renaming
+    rename_files(fmap_path, files, BIDSFMAPCase.TWO_PHASES_TWO_MAGNITUDES)
+
+
+def direct_fieldmaps_handler(fmap_path: Path):
+    sub = fmap_path.parent.parent.name
+    ses = fmap_path.parent.name
+
+    files = [f for f in os.listdir(fmap_path) if not f.startswith(".")]
+    # Assuming filename ends with "fmap" for the fieldmap
+    js = [f for f in files if f.endswith("fmap.json")][0]
+
+    # Checking for 'Units' key in fmap json
+    with open(fmap_path / js, "r") as file:
+        json_data = json.load(file)
+    if "Units" not in json_data:
+        cprint(
+            f'Assuming BIDS Case 3 : invalid file {js} for Direct Fieldmapping, missing "Units" key.',
+            lvl="warning",
+        )
+    else:
+        cprint(
+            f"Assuming BIDS Case 3 with 1 magnitude and 1 fieldmap files for {sub}, {ses}. Not supported yet.",
+            lvl="info",
+        )
+    not_supported_handler(fmap_path)
+
+
+def not_supported_handler(fmap_path: Path):
+    sub = fmap_path.parent.parent.name
+    ses = fmap_path.parent.name
+
+    to_delete = os.listdir(fmap_path)
+    cprint(
+        f"No currently supported BIDS case was recognized for {sub}, {ses}."
+        f"The following files will be deleted : {to_delete}.",
+        lvl="warning",
+    )
+    for file in to_delete:
+        os.remove(fmap_path / file)
+
+
+def infer_case_fmap(fmap_path: Path) -> BIDSFMAPCase:
+    files = [f for f in os.listdir(fmap_path) if not f.startswith(".")]
+    nb_files = len(files)
+
+    extensions = [
```

Comment:
You only need a set for `extensions`, so you can cast it to a Set here which would avoid doing it multiple times later.